README.md

```diff
@@ -97,6 +97,45 @@ get('/oauth2callback') do
 end
 ```
 
+### Example (Web with PKCE)
+
+Proof Key for Code Exchange (PKCE) is an [RFC](https://www.rfc-editor.org/rfc/rfc7636) that aims to prevent malicious operating system processes from hijacking an OAUTH 2.0 exchange. PKCE mitigates the above vulnerability by including `code_challenge` and `code_challenge_method` parameters in the Authorization Request and a `code_verifier` parameter in the Access Token Request.
+
+```ruby
+require 'googleauth'
+require 'googleauth/web_user_authorizer'
+require 'googleauth/stores/redis_token_store'
+require 'redis'
+
+client_id = Google::Auth::ClientId.from_file('/path/to/client_secrets.json')
+scope = ['https://www.googleapis.com/auth/drive']
+token_store = Google::Auth::Stores::RedisTokenStore.new(redis: Redis.new)
+authorizer = Google::Auth::WebUserAuthorizer.new(
+  client_id, scope, token_store, '/oauth2callback')
+
+
+get('/authorize') do
+  # NOTE: Assumes the user is already authenticated to the app
+  user_id = request.session['user_id']
+  # User needs to take care of generating the code_verifier and storing it in
+  # the session.
+  request.session['code_verifier'] ||= Google::Auth::WebUserAuthorizer.generate_code_verifier
```

Comment:
What is the reason for storing the `code_verifier` in the session? I understand the code was stateless so state needs to be stored elsewhere. 

In the example, does it make sense to illustrate where `code_verifier` is needed again?

Reply:
```What is the reason for storing the `code_verifier` in the session? I understand the code was stateless so state needs to be stored elsewhere.```
Yes, the reason behind storing `code_verifier` in the session is to make sure we maintain the state and pass on the same `code_verifier` in the subsequent call. Either we store the `code_verifier` itself in the session or we make sure that we're using the same `authorizer` object for further call, is upto the user how they want to design it.

```In the example, does it make sense to illustrate where `code_verifier` is needed again?```
     I have done that on Line [#123](https://github.com/googleapis/google-auth-library-ruby/pull/471/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R123). I can write a comment on top and explain it further if it's not clear. WDYT ?

Reply:
Hmm I think the disconnect for me was in this example, why have the step of storing it in the session, instead of just setting it directly

Is the code reading it from the session somewhere, or is the user supposed to read it from the session later?

Reply:
Okay got it, thank you!